pkg/operator/controller/externaldns/status_test.go

```diff
@@ -0,0 +1,681 @@
+			if cond.Type != tc.expectedResult.Type {
+				t.Errorf("expected condition type %v; got condition type %v", tc.expectedResult.Type, cond.Type)
+			}
+			if cond.Status != tc.expectedResult.Status {
+				t.Errorf("expected condition status %v; got condition status %v", tc.expectedResult.Status, cond.Status)
+			}
+			if cond.Reason != tc.expectedResult.Reason {
+				t.Errorf("expected condition reason %v; got condition reason %v", tc.expectedResult.Reason, cond.Reason)
+			}
+			if !strings.Contains(cond.Message, tc.expectedResult.Message) {
+				t.Errorf("expected condition message %v; got condition message %v", tc.expectedResult.Message, cond.Message)
+			}
+		})
+	}
+}
+
+func TestComputeAllReplicasCondition(t *testing.T) {
+	testCases := []struct {
+		name               string
+		existingDeployment appsv1.Deployment
+		expectedResult     metav1.Condition
+	}{
+		{
+			name:               "AvailableReplica = spec.replica should return ConditionTrue",
+			existingDeployment: fakeDeployment(appsv1.DeploymentAvailable, corev1.ConditionTrue, 8, "25%", "25%", 8, "external-dns-operator"),
+			expectedResult: metav1.Condition{
+				Type:    ExternalDNSDeploymentReplicasAllAvailableConditionType,
+				Status:  metav1.ConditionTrue,
+				Reason:  "DeploymentReplicasAvailable",
+				Message: "All replicas are available",
+			},
+		},
+		{
+			name:               "AvailableReplica < spec.replica shoud return ConditionFalse",
+			existingDeployment: fakeDeployment(appsv1.DeploymentProgressing, corev1.ConditionFalse, 8, "25%", "25%", 6, "external-dns-operator"),
+			expectedResult: metav1.Condition{
+				Type:    ExternalDNSDeploymentReplicasAllAvailableConditionType,
+				Status:  metav1.ConditionFalse,
+				Reason:  "DeploymentReplicasNotAvailable",
+				Message: "6/8 of replicas are available",
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			cond := computeAllReplicasCondition(&tc.existingDeployment)
+			if cond.Type != tc.expectedResult.Type {
+				t.Errorf("expected condition type %v; got condition type %v", tc.expectedResult.Type, cond.Type)
+			}
+			if cond.Status != tc.expectedResult.Status {
+				t.Errorf("expected condition status %v; got condition status %v", tc.expectedResult.Status, cond.Status)
+			}
+			if cond.Reason != tc.expectedResult.Reason {
+				t.Errorf("expected condition reason %v; got condition reason %v", tc.expectedResult.Reason, cond.Reason)
+			}
+			if !strings.Contains(cond.Message, tc.expectedResult.Message) {
+				t.Errorf("expected condition message %v; got condition message %v", tc.expectedResult.Message, cond.Message)
+			}
+		})
+	}
+}
+
+func TestComputeDeploymentPodsScheduledCondition(t *testing.T) {
+	testCases := []struct {
+		name               string
+		existingDeployment appsv1.Deployment
+		existingPods       []corev1.Pod
+		expectedResult     metav1.Condition
+	}{
+		//all podScheduledConditions are true
```

Comment:
These comments look redundant; the information is already conveyed in each test case's `name` field.